frontend/src/pages/ForgotPasswordPage.tsx

```diff
@@ -0,0 +1,251 @@
+import { ReactComponent as MailIcon } from '../icons/mail_icon.svg';
+import { api } from '../api/api';
+import css from './ForgotPassword.module.scss';
+
+const classes = classNames.bind(css);
+
+export const ForgotPasswordPage = () => {
+  const { t } = useTranslation();
+  const defaultDisableTime = 30; // Seconds
+
+  const [linkSent, setLinkSent] = useState(false);
+  const [email, setEmail] = useState('');
+  const [sendDisabled, setSendDisabled] = useState(false);
+  const [sendDisableTime, setSendDisableTime] = useState(
+    defaultDisableTime / 2,
+  );
+  const [resendDisabled, setResendDisabled] = useState(false);
+  const [resendDisableTime, setResendDisableTime] = useState(
+    defaultDisableTime,
+  );
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  let timeoutArray = useMemo(() => {
+    const array: NodeJS.Timeout[] = [];
+    return array;
+  }, []);
+
+  let intervalArray = useMemo(() => {
+    const array: NodeJS.Timeout[] = [];
+    return array;
+  }, []);
+
+  // Cleanup timers after unmounting
+  useEffect(
+    () => () => {
+      timeoutArray.forEach((timeout) => {
+        clearTimeout(timeout);
+      });
+      intervalArray.forEach((interval) => {
+        clearInterval(interval);
+      });
+    },
+    [timeoutArray, intervalArray],
+  );
+
+  const confirmationError = {
+    err: errorState(useState('')),
+  };
+
+  const sendEmailLink = () => {
+    return api.sendPasswordResetLink(email);
+  };
+
+  const createDisableTimer = (
+    setDisableTime: React.Dispatch<React.SetStateAction<number>>,
+    setDisabled: React.Dispatch<React.SetStateAction<boolean>>,
+    time: number,
+  ) => {
+    setDisabled(true);
+
+    const countdownInterval = setInterval(() => {
+      setDisableTime((previous) => previous - 1);
+    }, 1000);
+
+    const disableTimer = setTimeout(() => {
+      clearInterval(countdownInterval);
+
+      // Remove timeout and interval from the arrays after completing them
+      timeoutArray = timeoutArray.splice(timeoutArray.indexOf(disableTimer), 1);
+      intervalArray = intervalArray.splice(
+        intervalArray.indexOf(countdownInterval),
+        1,
+      );
+      setDisabled(false);
+      setDisableTime(time);
+    }, time * 1000);
+
+    timeoutArray.push(disableTimer);
+    intervalArray.push(countdownInterval);
+  };
+
+  const isUniqueError = (err: any) => err.response?.status === 404;
+
+  const handleSend = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    e.stopPropagation();
+    setIsLoading(true);
+
+    try {
+      await sendEmailLink();
+    } catch (err) {
+      if (isUniqueError(err)) {
+        confirmationError.err.setMessage(t('This email doesn’t exist'));
+      } else {
+        setErrorMessage(t('Email service error'));
+      }
+      setIsLoading(false);
+      return;
+    }
+
+    setResendDisableTime(defaultDisableTime / 2);
+    createDisableTimer(
```

Comment:
```suggestion
    createDisableTimer(
```

`createDisableTime()` could set the resend disable time, since the `time` parameter is just that